fastestimator/op/numpyop/univariate/rua.py

```diff
@@ -180,31 +180,35 @@ def __init__(self,
         super().__init__(inputs=inputs, outputs=outputs, mode=mode, num_bits=num_bits)
 
     def set_rua_level(self, magnitude_coef: float) -> None:
-        """Set the augmentation intentity based on the magnitude_coef.
+        """Set the augmentation intensity based on the magnitude_coef.
 
         This method is specifically designed to be invoked by the RUA Op.
 
         Args:
             magnitude_coef: The desired augmentation intensity (range [0-1]).
         """
-        if isinstance(self.num_bits, tuple):
-            if len(self.num_bits) == 3:
-                num_bits = []
-                for i in self.num_bits:
-                    if isinstance(i, tuple):
-                        num_bits.append(tuple([round(8-(magnitude_coef*i[1])), round(8-(magnitude_coef*i[0]))]))
-                    else:
-                        num_bits.append(round(8-(magnitude_coef*i)))
-                self.num_bits = tuple(num_bits)
-            else:
-                self.num_bits = round(8-(magnitude_coef*self.num_bits[1])), round(8-(magnitude_coef*self.num_bits[0]))
+        if isinstance(self.num_bits, tuple) and len(self.num_bits)==3:
+            num_bits = []
+            for i in self.num_bits:
+                num_bits.append(Posterize._range_tuple(num_bits=i, magnitude_coef=magnitude_coef))
+            self.num_bits = tuple(num_bits)
         else:
-            self.num_bits = round(8-(magnitude_coef*self.num_bits))
+            self.num_bits = Posterize._range_tuple(num_bits=self.num_bits, magnitude_coef=magnitude_coef)
         super().__init__(inputs=self.inputs,
                          outputs=self.outputs,
                          mode=self.mode,
                          num_bits=self.num_bits)
 
+    @staticmethod
+    def _range_tuple(num_bits: Union[int, Tuple[int, int]], magnitude_coef: float) -> Tuple[int, int]:
```

Comment:
we should have doc string for all non-override methods

Reply:
Sure. I'll add that.